src/main/DAL/MetaQuestion/MetaQuestionRepository.js

```diff
@@ -44,12 +80,78 @@ class MetaQuestionRepository {
      * @returns The updated question.
      */
     async addAnswersToQuestion(qId, answers) {
-        await answers.forEach(answer => {
+        answers.forEach(answer => {
             answer.metaQuestionId = qId;
         });
         await this.#Answer.bulkCreate(answers);
 
-        return await this.#MetaQuestion.findByPk(qId, {
+        return await this.getMetaQuestion(qId);
+    }
+
+    /**
+     * Adds given keywords to given meta-question.
+     * @param qId ID of an existing meta-question.
+     * @param keywords Array of keywords: string[]
+     * @returns The updated question.
+     */
+    async addKeywordsToQuestion(qId, keywords) {
+        const metaQuestion = await this.getMetaQuestion(qId);
+        const dbKeywords = await this.#getDbKeywords(keywords)
+        await metaQuestion.addKeywords(dbKeywords);
+        await metaQuestion.reload();
+        return metaQuestion;
+    }
+
+    /**
+     * Adds given keywords to given appendix.
+     * @param tag tag of an existing appendix.
+     * @param keywords Array of keywords: string[]
+     * @returns The updated appendix.
+     */
+    async addKeywordsToAppendix(tag, keywords) {
+        const appendix = await this.getAppendix(tag);
+        const dbKeywords = await this.#getDbKeywords(keywords)
+        await appendix.addKeywords(dbKeywords);
+        await appendix.reload();
+        return appendix;
+    }
+
+    /**
+     * @param tag Tag of the requested Appendix.
+     * @return The requested Appendix.
+     */
+    async getAppendix(tag) {
+        return await this.#Appendix.findByPk(tag, {
+            include: [
```

Comment:
amazing syntax

Reply:
This is sequelize, not me....
They did do an amazing job